### PR TITLE
[foobot] Foobot Binding initial contribution

### LINKED ISSUE #3247: [foobot] Add Foobot Binding
I am working on a new binding for the Indoor Air Quality Sensor "FOOBOT".
I will create the PR soon. 
The code for the binding is here - https://github.com/divyachauhan25/openhab2-addons

Expected Behavior -
Allow the sensor readings from the Foobot device to be displayed on the OpenHAB Control panel.

Current Behavior -
Binding does not exist.

My Environment -
Binding created for openHAB2-addons

### PR DESCRIPTION
**Pull request by Hilbrand**
This is the continued development of the original Foobot pr: #3266

Closes #3266
Closes #3247

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -45,6 +45,7 @@
 /bundles/org.openhab.binding.feed/ @svilenvul
 /bundles/org.openhab.binding.feican/ @Hilbrand
 /bundles/org.openhab.binding.folding/ @fa2k
+/bundles/org.openhab.binding.foobot/ @Hilbrand
```
[the remark this answers is not in the document]
Comment (Hilbrand) [reply CODEOWNERS]: Added @airboxlab as codeowner

File: bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotAccountHandler.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            return;
+        }
+        username = accountConfig.username;
+        connector.setApiKey(accountConfig.apiKey);
+        refreshInterval = accountConfig.refreshInterval;
+        if (this.refreshInterval < 5) {
+            logger.warn(
+                    "Refresh interval time [{}] is not valid. Refresh interval time must be at least 5 minutes. Setting to 7 minutes",
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotAccountHandler.java]: Interestingly, the constant's value is 8, not 7. Might be worth to use the constant as a parameter here.

File: bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotAccountHandler.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                    accountConfig.refreshInterval);
+            refreshInterval = DEFAULT_REFRESH_PERIOD_MINUTES;
+        }
+        logger.debug("Foobot Account bridge starting... user: {}, refreshInterval: {}", accountConfig.username,
+                refreshInterval);
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Wait to get associated devices");
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotAccountHandler.java]: CONFIGURATION_PENDING is not a valid sub-status of OFFLINE, see https://www.openhab.org/docs/concepts/things.html#status-details.

File: bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/discovery/FoobotAccountDiscoveryService.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    private @Nullable FoobotAccountHandler handler;
+    private @NonNullByDefault({}) ThingUID bridgeUID;
+
+    public FoobotAccountDiscoveryService() {
+        super(FoobotHandlerFactory.DISCOVERABLE_THING_TYPE_UIDS, TIMEOUT_SECONDS);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        retrieveFoobots();
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/discovery/FoobotAccountDiscoveryService.java]: Hm, this is doing a one-time discovery, not a (continuous) background discovery. I would claim that this DiscoveryService should not at all claim that it offers background discovery (add a `false` to the super call in the constructor).

File: bundles/org.openhab.binding.foobot/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+## Channels
+
+The bridge has one channel:
+
+| Channel ID           | Item Type | Description
+|----------------------|-----------|-----------------------------------------------
+| apiKeyLimitRemaining | Number    | The remaining number of API requests for today
+
+
+The AirQuality senors information that is retrieved is available as these channels:
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.foobot/README.md]:
```suggestion
The AirQuality sensors information that is retrieved is available as these channels:
```

File: bundles/org.openhab.binding.foobot/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 6]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="foobot"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Foobot Account -->
+	<bridge-type id="account">
+		<label>Foobot Account</label>
+		<description>Retrieves your Foobot account and all the devices associated with.</description>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.foobot/src/main/resources/ESH-INF/thing/thing-types.xml]:
```suggestion
		<description>Your Foobot account.</description>
```

File: bundles/org.openhab.binding.foobot/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+	</bridge-type>
+
+	<!-- Foobot Device -->
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="account" />
+		</supported-bridge-type-refs>
+
+		<label>Foobot</label>
+		<description>Shows the air quality data from the Foobot device.</description>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.foobot/src/main/resources/ESH-INF/thing/thing-types.xml]:
```suggestion
		<description>A Foobot device.</description>
```

**Comment by Hilbrand** [comment]
@kaikreuzer Thanks for merging. Can you add @airboxlab as a contributor (If you not already did this).

**Comment by kaikreuzer** [comment]
Done.

**Pull request merged**